python/ray/autoscaler/_private/aws/config.py

```diff
@@ -350,12 +350,23 @@ def _configure_key_pair(config):
         "Private key file {} not found for {}".format(key_path, key_name)
 
     config["auth"]["ssh_private_key"] = key_path
-    config["head_node"]["KeyName"] = key_name
-    config["worker_nodes"]["KeyName"] = key_name
+    for node_type in node_types:
+        node_config = node_types[node_type]["node_config"]
+        node_config["KeyName"] = key_name
```

Comment:
When doing something like this, we should explicitly raise a warning saying that KeyName in head_node/worker_nodes are deprecated and they must be passed in the node config.